### PR TITLE
refactor: replace semantic() with dsymbolSemantic()

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

**Review by JinShil — APPROVED** [review]

File: src/ddmd/aggregate.d [hunk 1]
```diff
@@ -21,6 +21,7 @@ import ddmd.declaration;
 import ddmd.dscope;
 import ddmd.dstruct;
 import ddmd.dsymbol;
+import ddmd.dsymbolsem;
```
Comment (wilzbach) [inline src/ddmd/aggregate.d]: You could have used a selective import here.

**Review by wilzbach — APPROVED** [review]

File: src/ddmd/aggregate.d (hunk 1 above)
Comment (WalterBright) [reply src/ddmd/aggregate.d]: 1. there'd not be much value in that, as dsymbolsem has only a handful of publics in it.

2. adding them now would bloat up these PRs by hundreds of lines.

3. I'm not real sure it is the right approach for dmd

Comment (ibuclaw) [reply src/ddmd/aggregate.d]: I think that while we should aim to reduce toplevel imports. Selective imports should probably be kept function local.

Eg: genCmain is the only user of the module ddmd.mars. Therefore it makes sense to remove the toplevel import of mars and encapsulate it into genCmain.

**Comment by ibuclaw** [comment]
It irks me that there's a lot of network related failures in our ci as of recent. @MartinNowak we have no logic for attempting actions in our scripts right? If I have time I'll look into that (ie: try 3 times to download file before failing).

**Comment by WalterBright** [comment]
@ibuclaw can we get this merged sooner? Thanks!

**Review by ibuclaw — APPROVED** [review]

**Comment by WalterBright** [comment]
Thanks!

**Comment by MartinNowak** [comment]
> It irks me that there's a lot of network related failures in our ci as of recent. @MartinNowak we have no logic for attempting actions in our scripts right? If I have time I'll look into that (ie: try 3 times to download file before failing).

We do have retries and fallback URLs in most places @ibuclaw. One problem is that there are too many CI systems which increases failure rates and makes debugging harder.

**Pull request merged**